src/tests/signatures.rs

```diff
@@ -156,29 +157,37 @@ fn test_rsa_signatures() {
         &mut testcase.result,
         &mut mechanism,
     );
-    assert_eq!(ret, CKR_OK);
+    #[cfg(any(feature = "fips", not(feature = "dynamic")))]
+    assert_ne!(ret, CKR_OK);
 
-    let result = match sig_gen(
-        session,
-        pri_key_handle,
-        &mut testcase.value,
-        &mut mechanism,
-    ) {
-        Ok(r) => r,
-        Err(e) => panic!("f{e}"),
-    };
-    assert_eq!(testcase.result, result);
+    /* SHA1 is disabled in out openssl submodule */
 
-    let result = match sig_gen_multipart(
-        session,
-        pri_key_handle,
-        &mut testcase.value,
-        &mut mechanism,
-    ) {
-        Ok(r) => r,
-        Err(e) => panic!("f{e}"),
-    };
-    assert_eq!(testcase.result, result);
+    #[cfg(all(not(feature = "fips"), feature = "dynamic"))]
```

Comment:
same here. I think the following should be equivalent:
```suggestion
    #[cfg(feature = "dynamic")]
```

Reply:
ah right ...